core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -110,21 +110,32 @@ public void commitState(Snapshot snapshot)
       throws CommitException, UnknownTransactionStatusException {
     String id = snapshot.getId();
     try {
-      commitState(snapshot.getId());
-    } catch (CoordinatorException e) {
-      TransactionState state = abort(id);
-      if (state.equals(TransactionState.ABORTED)) {
-        rollbackRecords(snapshot);
-        throw new CommitException(
-            "committing state in coordinator failed. the transaction is aborted", e);
+      Coordinator.State state = new Coordinator.State(id, TransactionState.COMMITTED);
+      coordinator.putState(state);
+      logger.debug(
+          "transaction {} is committed successfully at {}", id, System.currentTimeMillis());
+    } catch (CoordinatorConflictException e) {
+      try {
+        Optional<Coordinator.State> s = coordinator.getState(id);
```

Comment:
I think we should check the state only when catching `CoordinatorConflictException` because when we catch `CoordinatorException`, something unexpected database error is happening, so we probably can't get the state from the coordinator as well.

Reply:
I'm fine with this, but I think we could have `CoordinatorException` due to a temporary glitch, which is not necessarily an unexpected database error.